core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -45,6 +45,56 @@ trait EntityDecoder[F[_], T] { self =>
       override def consumes: Set[MediaRange] = self.consumes
     }
 
+  def recover(f: DecodeFailure => T)(implicit F: Functor[F]): EntityDecoder[F, T] =
```

Comment:
In cats, [recover](https://github.com/typelevel/cats/blob/c51fe610704aaa482c74b37b5ca2ec396749f5f3/core/src/main/scala/cats/ApplicativeError.scala#L64) takes a `PartialFunction` and [handleError](https://github.com/typelevel/cats/blob/c51fe610704aaa482c74b37b5ca2ec396749f5f3/core/src/main/scala/cats/ApplicativeError.scala#L37) takes a `Function1`.  It would be nice to remain consistent with that terminology, even if we don't have an `ApplicativeError` instance.

(And on that subject, could we? 🤔)

Reply:
Good point, I'll rename it to `handleError` if that's ok with you. Reason being, I prefer `Function1` over `PartialFunction`.

FYI, I tried constraining it to `ApplicativeError[F, DecodeFailure]` 

```scala
  def handleError(f: DecodeFailure => T)(
      implicit Ae: ApplicativeError[F, DecodeFailure]): EntityDecoder[F, T] =
    new EntityDecoder[F, T] {
      override def consumes: Set[MediaRange] = self.consumes

      override def decode(msg: Message[F], strict: Boolean): DecodeResult[F, T] =
        DecodeResult(Ae.handleError(self.decode(msg, strict).value)(e => Right(f(e))))
    }
```

and `core` compiled, but then realised that there is no `ApplicativeError[IO, DecodeFailure]` when I tried to compile tests. I ran into ambiguous implicits when I tried to add one via:

```scala
trait EntityDecoderInstances {
   implicit def applicativeError[F[_]](implicit F: Effect[F]): ApplicativeError[F, DecodeFailure] = new ApplicativeError[F, DecodeFailure] {
    override def raiseError[A](e: DecodeFailure): F[A] = F.raiseError(e)

    override def handleErrorWith[A](fa: F[A])(f: DecodeFailure => F[A]): F[A] = F.recoverWith(fa) {
      case decodeFailure: DecodeFailure => f(decodeFailure)
      case other => F.raiseError(other)
    }

    override def pure[A](x: A): F[A] = F.pure(x)

    override def ap[A, B](ff: F[A => B])(fa: F[A]): F[B] = F.ap(ff)(fa)
  }
}
```

Not sure how to approach it it we want to go down this route.

More info on that error:

```
[error]  both method applicativeError in trait lower of type [F[_]](implicit F: cats.effect.Effect[F])cats.ApplicativeError[F,org.http4s.DecodeFailure]
[error]  and value evidence$3 of type cats.effect.Effect[F]
[error]  match expected type cats.Functor[F]
[error]     DecodeResult.success(msg.body.chunks.runFoldMonoid)
[error]                         ^
[error] 
```